Non-resolvable Parent POM

The current master version of the API does not compile.
```
[ERROR]     Non-resolvable parent POM for org.eclipse.microprofile.metrics:microprofile-metrics-api:1.0-SNAPSHOT: Could not find artifact org.eclipse.microprofile.metrics:microprofile-metrics-parent:pom:0.1-SNAPSHOT and 'parent.relativePath' points at wrong local POM @ line 20, column 13 -> [Help 2]
```
Unless all modules are commented out and the POM is built separately the build does not work.

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -21,7 +21,7 @@
         <!-- This is just for now and will not work if the API has a separate release cycle than the rest. -->
         <groupId>org.eclipse.microprofile.metrics</groupId>
         <artifactId>microprofile-metrics-parent</artifactId>
-        <version>0.1-SNAPSHOT</version>
+        <version>1.0-SNAPSHOT</version>
     </parent>
 
     <artifactId>microprofile-metrics-api</artifactId>

